packages/devtools_app/lib/src/screens/memory/panes/chart/legend.dart

```diff
@@ -14,6 +14,102 @@ import 'memory_charts.dart';
 import 'memory_events_pane.dart';
 import 'memory_vm_chart.dart';
 
+class MemoryChartWidget extends StatelessWidget {
+  const MemoryChartWidget({
+    super.key,
+    required this.isAndroidVisible,
+    required this.chartController,
+  });
+
+  final bool isAndroidVisible;
+  final MemoryChartPaneController chartController;
+
+  @override
+  Widget build(BuildContext context) {
+    final theme = Theme.of(context);
+    final colorScheme = theme.colorScheme;
+    final legendHeading = theme.hoverTextStyle;
+
+    final legendRows = <Widget>[];
+
+    final events = eventLegendContent(colorScheme.isLight);
+    legendRows.add(
+      Container(
+        padding: legendTitlePadding,
+        child: Text('Events Legend', style: legendHeading),
+      ),
+    );
+
+    var iterator = events.entries.iterator;
+    while (iterator.moveNext()) {
+      final leftEntry = iterator.current;
+      final rightEntry = iterator.moveNext() ? iterator.current : null;
+      legendRows.add(
+        LegendRow(
+          entry1: leftEntry,
+          entry2: rightEntry,
+          chartController: chartController,
+        ),
+      );
+    }
+
+    final vms = vmLegendContent(chartController.vm);
+    legendRows.add(
+      Container(
+        padding: legendTitlePadding,
+        child: Text('Memory Legend', style: legendHeading),
+      ),
+    );
+
+    iterator = vms.entries.iterator;
+    while (iterator.moveNext()) {
+      final legendEntry = iterator.current;
+      legendRows.add(
+        LegendRow(
+          entry1: legendEntry,
+          chartController: chartController,
+        ),
+      );
+    }
+
+    if (isAndroidVisible) {
+      final androids = androidLegendContent(chartController.android);
+      legendRows.add(
+        Container(
+          padding: legendTitlePadding,
+          child: Text('Android Legend', style: legendHeading),
+        ),
+      );
+
+      iterator = androids.entries.iterator;
+      while (iterator.moveNext()) {
+        final legendEntry = iterator.current;
+        legendRows.add(
+          LegendRow(
+            entry1: legendEntry,
+            chartController: chartController,
+          ),
+        );
+      }
+    }
+
+    return Container(
+      width: legendWidth,
+      height: isAndroidVisible ? legendHeight2Charts : legendHeight1Chart,
+      padding: const EdgeInsets.fromLTRB(0, 5, 5, 8),
```

Comment:
use padding consts from theme.dart. denseSpacing, densePadding, etc.

Also, we shouldn't need bottom padding here